src/measure_theory/integral/circle_transform.lean

```diff
@@ -174,4 +173,308 @@ begin
   exact this,
 end
 
+/--Cauchy integral form of a function at `z` in a disk of radius `R`-/
+def circle_integral_form [complete_space E] (R : ℝ) (z : ℂ) (f : ℂ → E) : (ℂ → E) :=
+  λ w, (2 * π * I : ℂ)⁻¹ • (∮ z in C(z, R), (z - w)⁻¹ • f z)
+
+lemma circle_intgral_form_eq_int [complete_space E] (R : ℝ) (z : ℂ) (f : ℂ → E) :
+  circle_integral_form R z f = λ w,
+ ∫ (θ : ℝ) in 0..2 * π, (circle_transform R z w f) θ :=
+begin
+    simp_rw [circle_transform, circle_integral_form, circle_integral,
+      interval_integral.integral_smul],
+end
+
+lemma circle_transform_circle_int [complete_space E] (R : ℝ) (z w : ℂ) (f : ℂ → E) :
+  ∫ (θ : ℝ) in 0..2 * π, circle_transform R z w f θ =
+ (2 * π * I : ℂ)⁻¹ • ∮ z in C(z, R), (z - w)⁻¹ • f z :=
+begin
+  simp_rw [circle_transform, circle_integral,deriv_circle_map, circle_map],
+  simp only [real_smul, nsmul_eq_mul, nat.cast_bit0, nat.cast_one, one_div,
+  interval_integral.integral_smul, zero_add],
+end
+
+lemma circle_transform_has_deriv_at (R : ℝ) (z : ℂ) (f : ℂ → ℂ) :
+  ∀ (t : ℝ) y ∈ ball z R, has_deriv_at (λ y, (circle_transform R z y f) t)
+  ((circle_transform_deriv R z y f) t) y :=
+begin
+  intros y x hx,
+  simp only [circle_transform, circle_transform_deriv, algebra.id.smul_eq_mul,
+  ←mul_assoc, deriv_circle_map],
+  apply_rules [has_deriv_at.mul_const, has_deriv_at.const_mul],
+  have H : has_deriv_at (λ (y_1 : ℂ), (circle_map z R y - y_1)) (-1 ) x,
+  by {apply has_deriv_at.const_sub, apply has_deriv_at_id,},
+  have hfin := has_deriv_at.inv H (sub_ne_zero.2 (circle_map_ne_mem_ball hx y)),
+  simp only [one_div, neg_neg] at hfin,
+  apply hfin,
```

Comment:
```suggestion
  exact hfin,
```
Maybe there are some underscores missing.

General rule of thumb: no `apply` or `refine` in the last line of a proof.